tests/test_meta/test_grouped_transformer.py

```diff
@@ -0,0 +1,203 @@
+
+import pytest
+import numpy as np
+import pandas as pd
+from sklearn import clone
+from sklearn.linear_model import LinearRegression
+from sklearn.preprocessing import StandardScaler, MinMaxScaler
+from sklearn.utils import check_X_y
+
+from sklego.common import flatten
+from sklego.meta import GroupedTransformer
+from tests.conftest import transformer_checks, nonmeta_checks, general_checks
+
+
+@pytest.fixture
+def tests_to_skip():
+    return [
+        # Nonsense checks because we always need at least two columns (group and value)
+        "check_fit2d_1feature",
+        "check_fit2d_predict1d",
+    ]
+
+
+@pytest.mark.parametrize(
+    "test_fn", flatten([transformer_checks, nonmeta_checks, general_checks])
+)
+def test_estimator_checks(test_fn, tests_to_skip):
+    if test_fn.__name__ not in tests_to_skip:
+        trf = GroupedTransformer(StandardScaler())
+        test_fn(GroupedTransformer.__name__, trf)
+
+
+def test_values(random_xy_dataset_clf):
+    X, y = random_xy_dataset_clf
+    X, y = check_X_y(X, y)
+
+    # Make sure all groups are present
+    groups = np.repeat([0, 1], repeats=(X.shape[0] + 1) // 2)[:X.shape[0], np.newaxis]
+    X_with_groups = np.concatenate([groups, X], axis=1)
+
+    # Some weird interval to make sure we test the right values
+    scaling_range = (13, 42)
+
+    trf = MinMaxScaler(scaling_range)
+    transformer = GroupedTransformer(clone(trf), groups=0)
```

Comment:
I wonder ... why clone here?

Reply:
I think the first time I did this my brain was turned of. The other times I copied the line. Will get rid of it :)